cmp/grid/GridModel.ts

```diff
@@ -47,8 +77,248 @@ import {
+
+    /** Column ID(s) by which to do full-width grouping. */
+    groupBy?: Some<string>;
+
+    /** True (default) to show a count of group member rows within each full-width group row. */
+    showGroupRowCounts?: boolean;
+
+    /** Size of text in grid.  If undefined, will default and bind to `XH.sizingMode`. */
+    sizingMode?: SizingMode;
+
+    /** True to highlight the currently hovered row. */
+    showHover?: boolean;
+
+    /** True to render row borders. */
+    rowBorders?: boolean;
+
+    /** Specify treeMode-specific styling. */
+    treeStyle?: TreeStyle;
+
+    /** True to use alternating backgrounds for rows. */
+    stripeRow?: boolean;
+
+    /** True to render cell borders. */
+    cellBorders?: boolean;
+
+    /** True to highlight the focused cell with a border. */
+    showCellFocus?: boolean;
+
+    /** True to suppress display of the grid's header row. */
+    hideHeaders?: boolean;
+
+    /** True to disallow moving columns outside of their groups. */
+    lockColumnGroups?: boolean;
+
+    /** True to allow the user to manually pin / unpin columns via UI affordances. */
+    enableColumnPinning?: boolean;
+
+    /** True to enable exporting this grid and install default context menu items. */
+    enableExport?: boolean;
+
+    /** Default export options. */
+    exportOptions?: ExportOptions;
+
+    /**
+     * Closure to generate CSS class names for a row.
+     * NOTE that, once added, classes will *not* be removed if the data changes.
+     * Use `rowClassRules` instead if StoreRecord data can change across refreshes.
+     */
+    rowClassFn?: RowClassFn;
+
+    /**
+     * Object keying CSS class names to functions determining if they should be added or
+     * removed from the row. See Ag-Grid docs on "row styles" for details.
+     */
+    rowClassRules?: Record<string, RowClassRuleFn>;
+
+    /** Height (in px) of a group row. Note that this will override `sizingMode` for group rows. */
+    groupRowHeight?: number;
+
+    /** Function used to render group rows. */
+    groupRowRenderer?: GroupRowRenderer;
+
+    /**
+     * Function to use to sort full-row groups.  Called with two group values to compare
+     * in the form of a standard JS comparator.  Default is an ascending string sort.
+     * Set to `null` to prevent sorting of groups.
+     */
+    groupSortFn?: GridGroupSortFn;
+
+    /**
+     * Callback when a key down event is detected on the grid. Function will receive an event
+     * with the standard 'target' element. Note that the ag-Grid API provides limited ability to
+     * customize keyboard handling. This handler is designed to allow applications to workaround
+     * this.
+     */
+    onKeyDown?: (e: KeyboardEvent) => void;
+
+    /**
+     * Callback when a row is clicked - will receive an event with a data node containing
+     * the row's data. (Note that this may be null - e.g. for clicks on full-width group rows.)
+     */
+    onRowClicked?: (e: any) => void;
```

Comment:
Should we use `MouseEvent` for this and other click callbacks?

Reply:
these are actually agGrid event contexts that contain the native event.  Agree we want to strongly type these soon.